### ISSUE TITLE
PublishPipelineArtifact fails when special (socket) files are in dir

### ISSUE THREAD

**Issue by lvlie** [microsoft/azure-pipelines-tasks#12050]
## Required Information

**Question, Bug, or Feature?**  
*Type*: Bug

**Enter Task Name**: PublishPipelineArtifact 

## Environment
- Server - Azure Pipelines 
- Agent - Hosted (provided ubuntu-latest)

## Issue Description

Been troubleshooting the PublishPipelineArtifact after a "packer" build (not using the packer task, but plain qemu based build) on a hosted agent for quite some time. The task output did not point me to anything useful. Only when trying to copy the artifacts to a staging dir did the CopyFiles task provide some useful info. 

### Task logs

```
##[debug]Evaluating condition for step: 'Publish base image'
##[debug]Evaluating: SucceededNode()
##[debug]Evaluating SucceededNode:
##[debug]=> True
##[debug]Result: True
Starting: Publish base image
==============================================================================
Task         : Publish Pipeline Artifacts
Description  : Publish (upload) a file or directory as a named artifact for the current run
Version      : 1.2.1
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/publish-pipeline-artifact
==============================================================================
Artifact name input: base-image
Uploading pipeline artifact from /home/vsts/work/1/s/packer/ for build #23077
##[debug]Dedup parallelism: 192
Information, ApplicationInsightsTelemetrySender will correlate events with X-TFS-Session d0638e16-7140-4e8b-979f-b700d7d508c0
Information, DedupManifestArtifactClient will correlate http requests with X-TFS-Session d0638e16-7140-4e8b-979f-b700d7d508c0
Information, 2 files to be processed in 1 groups.
Error, Failed to hash a page of files. Exception listed below...
Error, System.IO.IOException: No such device or address
   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
   at Microsoft.VisualStudio.Services.Content.Common.FileStreamUtils.OpenFileStreamForAsync(String filePath, FileMode mode, FileAccess fileAccess, FileShare fileShare, FileOptions extraOptions)
   at Microsoft.VisualStudio.Services.BlobStore.Common.ChunkerHelper.CreateFromFileAsync(IFileSystem fileSystem, String path, CancellationToken cancellationToken, Boolean configureAwait)
   at Microsoft.VisualStudio.Services.BlobStore.Common.FileBlobDescriptor.CalculateAsync(IFileSystem fileSystem, String rootDirectory, Boolean chunkDedup, String relativePath, FileBlobType fileBlobType, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.BlobStore.Common.PrecomputedHashesGenerator.GetFileBlobDescriptorAsync(String rootDirectory, Boolean chunkDedup, PageItem pageItem, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.BlobStore.Common.PrecomputedHashesGenerator.<>c__DisplayClass25_0.<<PaginateAndProcessFilesHelperAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Services.Content.Common.NonSwallowingActionBlock.<>c__DisplayClass5_0`1.<<CreateNonSwallowingFunc>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Services.BlobStore.Common.PrecomputedHashesGenerator.PaginateAndProcessFilesHelperAsync(String sourceDirectory, Boolean chunkDedup, IEnumerable`1 pathsInDir, Action`1 hashCompleteCallback, Int32 currentPage, Int32 totalPages, CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.BlobStore.Common.PrecomputedHashesGenerator.<>c__DisplayClass22_0.<<PaginateAndProcessFiles>b__1>d.MoveNext()
Information, ApplicationInsightsTelemetrySender correlated 2 events with X-TFS-Session d0638e16-7140-4e8b-979f-b700d7d508c0
##[error]No such device or address
##[debug]Processed: ##vso[task.logissue type=error;]No such device or address
##[debug]Processed: ##vso[task.complete result=Failed;]
##[debug]   at Interop.ThrowExceptionForIoErrno(ErrorInfo errorInfo, String path, Boolean isDirectory, Func`2 errorRewriter)
   at Microsoft.Win32.SafeHandles.SafeFileHandle.Open(String path, OpenFlags flags, Int32 mode)
```

The error points in a very general direction, but it took quite some builds and tries to figure out what the issue was. 
Only after using the CopyFile task did I get the following output:
```
##[debug]cp failed
##[debug]Failed cp: cp: copyFileSync: could not read src file (/home/vsts/work/1/s/packer/image.qcow2.monitor)
```
Which is a QMP socket file (with srwxr-xr-x) made and left over by packer (for which I'll create a separate issue with hashicorp). Makes sense that PublishArtifact goes wrong on such a file, but it should exit with a neater, more explanatory error message, imo. 

### Error logs
```
Error, Failed to hash a page of files. Exception listed below...
Error, System.IO.IOException: No such device or address
```

Hope this helps making the task even better.
